src/App.tsx

```diff
@@ -14,11 +15,114 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[]
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  chooseGoodsHandler = (item: string) => {
+    const { selectedGoods } = this.state;
+
+    if (!(selectedGoods.includes(item))) {
+      this.setState(prevState => ({
+        selectedGoods: [...prevState.selectedGoods, item],
+      }));
+    }
+  };
+
+  RemoveGoodsHandler = (item: string) => {
+    this.setState(prevState => ({
+      selectedGoods:
+       prevState.selectedGoods.filter(element => element !== item),
+    }));
+  };
+
+  resetSelectedGoods() {
+    this.setState({
+      selectedGoods: [],
+    });
+  }
+
+  showGoodsHandler() {
+    const { selectedGoods } = this.state;
+
+    switch (selectedGoods.length) {
+      case 0:
+        return 'No goods selected';
+
+      case 1:
+        return `${selectedGoods[0]} is selected`;
+
+      case 2:
+        return `${selectedGoods.join(' and ')} are selected`;
+
+      default:
+        return `${selectedGoods.slice(0, -1).join(', ')} and ${selectedGoods[selectedGoods.length - 1]} are selected`;
```

Comment:
є чудовий метод at, який бере елемент з масиву і може працювати з від`ємними числами
```suggestion
        return `${selectedGoods.slice(0, -1).join(', ')} and ${selectedGoods.at(-1)} are selected`;
```